src/debug.cpp

```diff
@@ -74,6 +76,13 @@ void fatal_error_fn(const char *msg, const char *file,
 	abort();
 }
 
+std::string debug_describe_exc(const std::exception &e)
+{
+	if (dynamic_cast<const std::bad_alloc*>(&e))
+		return "C++ out of memory";
```

Comment:
Maybe include the original error message in case it contains something useful? (But I don't know if that can actually happen, in my tests `what()` always just returned `"std::bad_alloc"`.)

```suggestion
		return std::string("C++ out of memory").append(" (\"").append(e.what()).append("\")");
```

Reply:
In theory someone could subclass `std::bad_alloc` but I think in practice that's not a thing either.

Reply:
I took a look at the C++ standard. The string returned by `what()` is implementation-defined, so in theory it could be something other than the class name.

But this is fine by me, too.